File: TDDMicroExercises.ProposedSolution/Java/TirePressureMonitoringSystem/src/main/java/tddmicroexercises/tirepressuremonitoringsystem/Sensor.java
package tddmicroexercises.tirepressuremonitoringsystem;

import java.util.Random;

public class Sensor implements ISensor
{
    public static final double OFFSET = 16;

    public double popNextPressurePsiValue()
    {
        double pressureTelemetryValue;
        pressureTelemetryValue = samplePressure( );

        return OFFSET + pressureTelemetryValue;
    }

    private static double samplePressure()
    {
        // placeholder implementation that simulate a real sensor in a real tire
        Random basicRandomNumbersGenerator = new Random(42);
        double pressureTelemetryValue = 6 * basicRandomNumbersGenerator.nextDouble() * basicRandomNumbersGenerator.nextDouble();
        return pressureTelemetryValue;
    }
}

File: Swift/TirePressureMonitoringSystem/TirePressureMonitoringSystem/TirePressureMonitoringSystemTests/Sensor.swift
import Foundation

class Sensor {

    static let offset = 16.0

    func popNextPressurePsiValue() -> Double {
        let pressureTelemetryValue = Sensor.samplePressure()
        return Sensor.offset + pressureTelemetryValue
    }

    private static func samplePressure() -> Double {
        arc4random()
        // placeholder implementation that simulate a real sensor in a real tire
        return 6 * nextDouble() * nextDouble()
    }

    private static func nextDouble() -> Double {
        return Double(arc4random())/Double(UInt32.max)
    }
}
